src/main/java/org/candlepin/async/tasks/CloudAccountOrgSetupJob.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.async.tasks;
+
+import org.candlepin.async.ArgumentConversionException;
+import org.candlepin.async.AsyncJob;
+import org.candlepin.async.JobArguments;
+import org.candlepin.async.JobConfig;
+import org.candlepin.async.JobConfigValidationException;
+import org.candlepin.async.JobConstraints;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+import org.candlepin.service.CloudProvider;
+import org.candlepin.service.CloudRegistrationAdapter;
+import org.candlepin.service.exception.CloudAccountOrgSetupException;
+import org.candlepin.service.model.CloudAccountData;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+public class CloudAccountOrgSetupJob implements AsyncJob {
+
+    private static final Logger log = LoggerFactory.getLogger(CloudAccountOrgSetupJob.class);
+
+    public static final String JOB_KEY = "CloudAccountOrgSetupJob";
+    public static final String JOB_NAME = "Cloud Account Org Setup Job";
+
+    protected static final String CLOUD_ACCOUNT_ID = "cloud_account_id";
+    protected static final String OFFERING_ID = "offering_id";
+    protected static final String CLOUD_PROVIDER = "cloud_provider";
```

Comment:
I know I asked you to remove the optional owner key from here before, but after more discussions it turns out we need to add it back :/

Reply:
done